pkg/converter/cloudevents/cloudevents.go

```diff
@@ -140,13 +141,22 @@ func (ce *CloudEvent) Request(request []byte, headers http.Header) ([]byte, map[
 		return request, nil, nil
 	}
 
+	clientContext, err := json.Marshal(map[string]interface{}{
+		"custom": context,
+	})
+
+	if err != nil {
+		return nil, nil, fmt.Errorf("cannot encode request client context: %w", err)
+	}
+
 	ceContext, err := json.Marshal(context)
```

Comment:
Any ideas about avoiding a double marshal of the context?

Reply:
I have one idea, but it is not for everyone - we can skip marshaling client context and just do this:

```go
	runtimeContext := map[string]string{
		ClientContextKey: fmt.Sprintf("{\"custom\":%s}", ceContext),
		CeContextKey:     string(ceContext),
	}
```

I'll let you decide if this is better than the double marshaling 🙂

Reply:
Yeah, I thought about something like that but thought I'd keep my dirty tricks for another time :grin: I've pushed this change !